### PR TITLE
Add missing environment variables to `build-controller.sh`

### PR DESCRIPTION
**Pull request by A-Hilaly**
Issue N/A

Description of changes:

Merge `build-controller.sh` and `Makefile` changes from https://github.com/aws-controllers-k8s/code-generator/commit/92458ababbeb0cd0f00b93e149ccc6b4e42f2835:
- Add missing environment variables `AWS_SDK_GO_VERSION` and `TEMPLATES_DIR`  to `build-controller.sh` script
- Add `-tags=codegen` arg to the go install ack-generate command
- Remove `build-ack-generate` step from `build-controller` step

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by A-Hilaly** [comment]
@surajkota Once this PR is merged you will be able to specify `AWS_SDK_GO_VERSION` when building a service controller from the community repo

File: Makefile
Comment (jaypipes) [inline Makefile:55]: any reason to remove this prerequisite?

Comment (A-Hilaly) [reply Makefile:55]: I thought that it will not work for contributors that installed `ack-generate` using `go get` but apparently it will ... https://github.com/aws-controllers-k8s/community/blob/main/Makefile#L33

Comment (jaypipes) [reply Makefile]: @A-Hilaly you're up way too late :)

If you can add the build-controller target back in, that would be great. I'll approve this after.

**Review by jaypipes — APPROVED** [review]

**Pull request merged**
